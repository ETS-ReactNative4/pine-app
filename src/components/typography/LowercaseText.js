import React from 'react';
import { StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  lowercase: {
    textTransform: 'lowercase'
  }
});

const LowercaseText = (props) => (
  <Text style={styles.lowercase}>
    {props.children}
  </Text>
);

LowercaseText.propTypes = {
  children: PropTypes.node
};

export default LowercaseText;
