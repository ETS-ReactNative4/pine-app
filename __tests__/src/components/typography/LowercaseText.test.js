import React from 'react';
import renderer from 'react-test-renderer';
import LowercaseText from '../../../../src/components/typography/LowercaseText';

describe('LowercaseText', () => {
  it('renders text as lowercase', () => {
    const tree = renderer.create(
      <LowercaseText>LOWERCASE</LowercaseText>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
