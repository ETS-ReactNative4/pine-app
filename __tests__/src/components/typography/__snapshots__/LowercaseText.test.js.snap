// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LowercaseText renders text as lowercase 1`] = `
<Text
  style={
    Object {
      "textTransform": "lowercase",
    }
  }
>
  LOWERCASE
</Text>
`;
